### PR TITLE
Allows both OL and UL lists for tab buttons

### PR DESCRIPTION
**Pull request by tiesont**
Modifies tab.js to allow OL as a parent element for the tab list items.

Duplicates UL-based visual tests to provide OL-based versions.

Adds check for "show" class to tabs unit tests.

### CODE REVIEW

File: js/tests/unit/tab.js [hunk 1]
```diff
@@ -71,13 +73,15 @@ $(function () {
...
     $(pillsHTML).find('li:last-child a').bootstrapTab('show')
     assert.strictEqual($('#qunit-fixture').find('.active').attr('id'), 'profile')
+    assert.strictEqual($('#qunit-fixture').find('.show').attr('id'), 'profile')
 
     $(pillsHTML).find('li:first-child a').bootstrapTab('show')
     assert.strictEqual($('#qunit-fixture').find('.active').attr('id'), 'home')
+    assert.strictEqual($('#qunit-fixture').find('.show').attr('id'), 'home')
   })
 
   QUnit.test('should activate element by tab id in ordered list', function (assert) {
```
Comment (Johann-S) [inline js/tests/unit/tab.js]: it's the only unit test needed here, the other aren't related

**Review by Johann-S — CHANGES_REQUESTED** [review]
Some of your changes are irrelevant

**Comment by XhmikosR** [comment]
@tiesont: can you take care of @Johann-S's comment?

**Comment by XhmikosR** [comment]
Closing due to lack of response. You can comment here and we can re-open it if you decide to resume working on the requested changes.

**Comment by tiesont** [comment]
I can fix this, I suppose, but I'm not sure what exactly is the issue here - I added those tests to make sure the plugin actually works with both UL and OL markup, which was the whole point of the original issue. Should I assume what you want is just the 

```javascript
assert.strictEqual($('#qunit-fixture').find('.show').attr('id'), 'home')
```

test? Or, do you only want the tests in tab.js?

**Comment by Johann-S** [comment]
That's right, you just have to create one new test, with `<ol>` list, which show it works 👍

**Comment by tiesont** [comment]
One _unit_ test (which should already be there) or one _visual_ test (which is also already there)? Is the issue that I modified the existing visual test to indicate which was UL-based and which was OL-based?

**Comment by Johann-S** [comment]
@tiesont the is, you have unrelated changes.
Your fix should be:

image

Because ordered list works, but with your fix we use the correct selector, so the use case is already covered

**Comment by tiesont** [comment]
Okay, I'll admit that I'm not all that great at using Git (or GitHub, for that matter). I pulled down the 4.1.3 source, merged the updated tab.js file, and then made the updates to allow OL as a selector. I reverted the changes I made to the unit test and visual test files to match what is in this branch.

**Comment by Johann-S** [comment]
I updated your branch and removed unrelated changes.

For me we should add one thing, it's a little note on our documentation to show we accept `<ol>` too 🤔 
You can edit that file to do that: https://github.com/twbs/bootstrap/blob/v4-dev/site/docs/4.1/components/navs.md#base-nav

**Review by Johann-S — APPROVED** [review]

**Comment by Johann-S** [comment]
@XhmikosR do you know why Github tells me there are conflicts here ? I resolved all the conflicts 😟

**Comment by XhmikosR** [comment]
I don't see any conflicts here.

**Comment by XhmikosR** [comment]
Ah, I see it now. It's shown if you switch Rebase and merge. Because there's a merge commit.

**Comment by Johann-S** [comment]
Thank you @tiesont 👍

**Comment by tiesont** [comment]
@Johann-S No problem. Thank you for being patient with me.

**Pull request merged**